### PR TITLE
wallet_rpc_server: add a validate_address RPC

### PR DESCRIPTION
**Pull request by moneromooo-monero**

### CODE REVIEW

File: src/wallet/wallet_rpc_server_commands_defs.h [hunk 1]
```diff
@@ -2189,5 +2189,40 @@ namespace wallet_rpc
...
+      bool any_net_type;
+      bool allow_openalias;
+
+      BEGIN_KV_SERIALIZE_MAP()
+        KV_SERIALIZE(address)
+        KV_SERIALIZE_OPT(any_net_type, false)
+        KV_SERIALIZE_OPT(allow_openalias, false)
+      END_KV_SERIALIZE_MAP()
+    };
+    typedef epee::misc_utils::struct_init<request_t> request;
```
Comment (stoffu) [inline src/wallet/wallet_rpc_server_commands_defs.h]: Looks like this class template isn't found anywhere?
```
/home/buildbot/builder/monero-static-ubuntu-amd64/build/src/wallet/wallet_rpc_server_commands_defs.h:2206:31: error: 'struct_init' in namespace 'epee::misc_utils' does not name a template type
     typedef epee::misc_utils::struct_init<request_t> request;
                               ^~~~~~~~~~~
```

Comment (moneromooo-monero) [reply src/wallet/wallet_rpc_server_commands_defs.h]: Looks like it's in #5101

**Comment by jtgrassie** [comment]
Shouldn't you bump the minor version in `wallet_rpc_server_commands_defs.h`?

File: src/wallet/wallet_rpc_server_commands_defs.h (hunk 1 above)
Comment (TheCharlatan) [reply src/wallet/wallet_rpc_server_commands_defs.h]: Builds fine when rebased on #5101. Note that there is a small merge conflict in p2p/net_node.h

**Comment by krtschmr** [comment]
@SmajeNz0 open a new issue for this, rather than asking in a merged pull

**Comment by SmajeNz0** [comment]
It has been fixed since a long time by monermoo.
Cheers

**Pull request merged**